### PR TITLE
[WIP] Performance test comparison mode

### PR DESCRIPTION
**Pull request by dimarub2000**
I hereby agree to the terms of the CLA available at: https://yandex.ru/legal/cla/?lang=en

For changelog. Remove if this is non-significant change.

Category (leave one):
- New Feature

Short description (up to few sentences):
#6478

### CODE REVIEW

**Comment by dimarub2000** [comment]
https://a.yandex-team.ru/review/971079/details

**Comment by dimarub2000** [comment]
Currently working here https://nda.ya.ru/3VnmJf

File: dbms/programs/performance-test/PerformanceTest.cpp [hunk 1]
```diff
@@ -257,90 +282,138 @@ std::vector<TestStats> PerformanceTest::execute()
         return statistics_by_run;
     }
 
-    // Pull memory usage data from query log. The log is normally filled in
-    // background, so we have to flush it synchronously here to see all the
-    // previous queries.
+    for (auto & connection : connections)
     {
```
Comment (alesapin) [inline dbms/programs/performance-test/PerformanceTest.cpp]: Two nested blocks and one of them is empty?

File: dbms/programs/performance-test/PerformanceTest.cpp [hunk 2]
```diff
@@ -257,90 +282,138 @@ std::vector<TestStats> PerformanceTest::execute()
...
-        TestStopConditions & stop_conditions = test_info.stop_conditions_by_run[run_index];
-        TestStats & statistics = statistics_by_run[run_index];
-        statistics.startWatches();
+        LOG_INFO(log, "[" << run_index << "] Run query '" << query << "'");
+
+        TestStopConditions & stop_conditions_by_connections = test_info.stop_conditions_by_run[run_index];
+        TestStats & statistics_by_connections = statistics_by_run[run_index];
+        t_test.clear();
+
+        pcg64 generator(randomSeed());
```
Comment (alesapin) [inline dbms/programs/performance-test/PerformanceTest.cpp]: Why do we need to create it on each iteration?

**Comment by qoega** [comment]
@akuzm made first version in #6478

**Pull request closed without merging**